packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1248,15 +1248,14 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
     MacOSPlugin.kConfigKey,
     WindowsPlugin.kConfigKey,
   ];
-  final Map<String, List<PluginInterfaceResolution>> possibleResolutions =
-      <String, List<PluginInterfaceResolution>>{};
-  final Map<String, String> defaultImplementations = <String, String>{};
-  // Generates a key for the maps above.
-  String getResolutionKey({required String platform, required String packageName}) {
-    return '$packageName:$platform';
-  }
```

Comment:
The variables and the method is not needed any more, cause it just unifies the resolution to plugin in platform on a more global scope. As the maps had a unique (`plugin:platform`) key, it is equal to processing them per platform in the for `platform` loop and just use the plugin package name as the key.